Migrate docs to Wasp Spec (#4191)
wasp-lang/wasp: web/docs/tutorial/02-project-structure.md

Reviewer: Should we link these to the API reference?

However, since this tutorial contains only a handful of files, there's no need for fancy organization.
We'll keep it simple by placing everything in the root `src` directory.

Many other files (e.g., `tsconfig.json`, `tsconfig.src.json`, `tsconfig.wasp.json`, `vite-env.d.ts`, etc.) help Wasp and the IDE improve your development experience with autocompletion, IntelliSense, and error reporting.

The `vite.config.ts` file is used to configure [Vite](https://vitejs.dev/guide/), Wasp's build tool of choice.

> We won't be configuring Vite in this tutorial, so you can safely ignore the file. Still, if you ever end up wanting more control over Vite, you'll find everything you need to know in [custom Vite config docs](../project/custom-vite-config.md).

Reviewer: ```suggestion
We won't be customizing the Vite setup in this tutorial, so you can safely ignore the file. Still, if you ever end up wanting more control over Vite, you'll find everything you need to know in [custom Vite config docs](../project/custom-vite-config.md).
```

The `schema.prisma` file is where you define your database schema using [Prisma](https://www.prisma.io/). We'll cover this a bit later in the tutorial.

The most important file in the project is `main.wasp.ts`. Wasp uses the configuration within it to perform its magic. Based on what you write, it generates a bunch of code for your database, server-client communication, React routing, and more.

Let's take a closer look at `main.wasp.ts`

## `main.wasp.ts`

`main.wasp.ts` is your app's Wasp file.
It defines the app's central components and helps Wasp to do a lot of the legwork for you.

> The file exports your app's top-level configuration and a `parts` array. Each entry defines a route, page, Query, Action, or another part of your app.

Reviewer: ```suggestion
The file exports your app's top-level configuration and a `parts` array. Each entry defines a Route, Page, Query, Action, or other features provided by Wasp.
```
